repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -97,13 +100,25 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
   }
 
   /**
-    * Gets the SAMl response from the post encoded body, and decodes it into a plain String.
+    * Gets the SAMl response from the post encoded body, and decodes it.
     *
     * @param request the servlet request that has an encoded body
     * @return the decoded saml response
     * @throws SamlPolicyException if decoding fails
     */
-  def decodeSamlResponse(request: HttpServletRequest): InputStream = ???
+  def decodeSamlResponse(request: HttpServletRequest): InputStream = {
+    try {
+      Option(request.getParameter("SAMLResponse"))
```

Comment:
Will the `getParameter` method properly handle the URL encoding of the value?

For example, would this work if the request coming in the door had parameters that looked like this:

```
SAMLResponse=PHNhbWxwOkxvZ291dFJlc3BvbnNlIHhtbG5zOnNhbWxwPSJ1cm46b2FzaXM6bmFtZXM6dGM6U0FNTDoyLjA6cHJvdG9jb2wiIHhtbG5zPSJ1cm46b2FzaXM6bmFtZXM6dGM6U0FNTDoyLjA6YXNzZXJ0aW9uIg0KICAgIElEPSJiMDczMGQyMWI2MjgxMTBkOGI3ZTAwNDAwNWIxM2EyYiIgSW5SZXNwb25zZVRvPSJkMmI3YzM4OGNlYzM2ZmE3YzM5YzI4ZmQyOTg2NDRhOCINCiAgICBJc3N1ZUluc3RhbnQ9IjIwMDQtMDEtMjFUMTk6MDA6NDlaIiBWZXJzaW9uPSIyLjAiPg0KICAgIDxJc3N1ZXI%2BaHR0cHM6Ly9TZXJ2aWNlUHJvdmlkZXIuY29tL1NBTUw8L0lzc3Vlcj4NCiAgICA8c2FtbHA6U3RhdHVzPg0KICAgICAgICA8c2FtbHA6U3RhdHVzQ29kZSBWYWx1ZT0idXJuOm9hc2lzOm5hbWVzOnRjOlNBTUw6Mi4wOnN0YXR1czpTdWNjZXNzIi8%2BDQogICAgPC9zYW1scDpTdGF0dXM%2BDQo8L3NhbWxwOkxvZ291dFJlc3BvbnNlPg%3D%3D&RelayState=token
```